library/src/scala/runtime/DynamicTuple.scala

```diff
@@ -264,6 +264,68 @@ object DynamicTuple {
         .asInstanceOf[Map[This, F]]
   }
 
+  def dynamicTake[This <: Tuple, N <: Int](self: This, n: N): Take[This, N] = {
+    if (n < 0) throw new IndexOutOfBoundsException(n.toString)
+    val actualN = Math.min(n, self.size)
+
+    type Result = Take[This, N]
+    val arr = (self: Any) match {
+      case () => Array.empty[Object]
+      case xxl: TupleXXL =>
+        xxl.elems.asInstanceOf[Array[Object]].take(actualN)
+      case _ =>
+        val arr = new Array[Object](actualN)
+        self.asInstanceOf[Product].productIterator.asInstanceOf[Iterator[Object]]
+          .copyToArray(arr, 0, actualN)
+        arr
+    }
+    dynamicFromIArray(arr.asInstanceOf[IArray[Object]]).asInstanceOf[Result]
+  }
+
+  def dynamicDrop[This <: Tuple, N <: Int](self: This, n: N): Drop[This, N] = {
+    if (n < 0) throw new IndexOutOfBoundsException(n.toString)
+    val size = self.size
+    val actualN = Math.min(n, size)
+
+    type Result = Drop[This, N]
+    val arr = (self: Any) match {
+      case () => Array.empty[Object]
+      case xxl: TupleXXL =>
+        xxl.elems.asInstanceOf[Array[Object]].drop(actualN)
+      case _ =>
+        val rem = size - actualN
+        val arr = new Array[Object](rem)
```

Comment:
```scala
        if rem == 0 then  
          Array.empty[Object]
        else 
          val arr = new Array[Object](rem)
          ...
```

Reply:
Use `Array.emptyObjectArray` or something like that

Reply:
Yeah, moving that to the default case seems a good idea, I'll do this.

Reply:
I didn't do it in `split` though, because I think it would add more branches than necessary.

By the way, removing the Unit case in the pattern matches might not really increase the speed, since most of the times we enter the Tuple1 to Tuple22 case, in which we added a new branch.

Reply:
The initial type test for `Unit` already adds this brach. We would just move it a bit later.